cmd/state-svc/test/integration/svc_int_test.go

```diff
@@ -207,40 +208,61 @@ func (suite *SvcIntegrationTestSuite) TestAutostartConfigEnableDisable() {
 	ts := e2e.New(suite.T(), false)
 	defer ts.Close()
 
+	autostartDir := filepath.Join(ts.Dirs.Config, "autostart")
+	err := fileutils.Mkdir(autostartDir)
+	suite.Require().NoError(err)
+	err = os.Setenv("_TEST_AUTOSTART_DIR", autostartDir)
+	suite.Require().NoError(err)
+	defer os.Unsetenv("_TEST_AUTOSTART_DIR")
+
+	if runtime.GOOS == "linux" {
+		err = fileutils.Touch(filepath.Join(autostartDir, ".profile"))
+		suite.Require().NoError(err)
+	}
+
 	cfg, err := config.New()
 	suite.Require().NoError(err)
 	as, err := autostart.New(svcAutostart.App, ts.SvcExe, nil, svcAutostart.Options, cfg)
 	suite.Require().NoError(err)
-	enabled, err := as.IsEnabled() // checks if the proper files are in place, not the config key setting
-	suite.Require().NoError(err)
 
+	var enabled bool
 	// Toggle it via state tool config.
-	cp := ts.SpawnWithOpts(e2e.WithArgs("config", "set", constants.AutostartSvcConfigKey, strconv.FormatBool(!enabled)))
+	cp := ts.SpawnWithOpts(
+		e2e.WithArgs("config", "set", constants.AutostartSvcConfigKey, strconv.FormatBool(!enabled)),
+		e2e.AppendEnv(fmt.Sprintf("_TEST_AUTOSTART_DIR=%s", autostartDir)),
+	)
 	cp.ExpectExitCode(0)
-	suite.checkEnabled(as, !enabled)
+	suite.Require().True(suite.expectEnabled(as, !enabled), ts.DebugMessage(fmt.Sprintf("autostart should be %v", !enabled)))
 
 	// Toggle it again via state tool config.
-	cp = ts.SpawnWithOpts(e2e.WithArgs("config", "set", constants.AutostartSvcConfigKey, strconv.FormatBool(enabled)))
+	cp = ts.SpawnWithOpts(
+		e2e.WithArgs("config", "set", constants.AutostartSvcConfigKey, strconv.FormatBool(enabled)),
+		e2e.AppendEnv(fmt.Sprintf("_TEST_AUTOSTART_DIR=%s", autostartDir)),
+	)
 	cp.ExpectExitCode(0)
-	suite.checkEnabled(as, enabled)
+	suite.Require().True(suite.expectEnabled(as, enabled), ts.DebugMessage(fmt.Sprintf("autostart should be %v", enabled)))
 }
 
 type autostartApp interface {
 	IsEnabled() (bool, error)
 }
 
-func (suite *SvcIntegrationTestSuite) checkEnabled(as autostartApp, expect bool) {
+func (suite *SvcIntegrationTestSuite) expectEnabled(as autostartApp, expect bool) bool {
 	timeout := time.After(1 * time.Minute)
 	tick := time.Tick(1 * time.Second)
 	for {
 		select {
 		case <-timeout:
 			suite.Fail("autostart has not been changed")
+			return false
 		case <-tick:
 			toggled, err := as.IsEnabled()
-			suite.Require().NoError(err)
-			if suite.Equal(expect, toggled) {
-				return
+			if err != nil {
+				suite.FailNow("failed to check autostart", err.Error())
+				return false
+			}
+			if expect == toggled {
+				return true
```

Comment:
Why not do the assertion here? You have failNow above, so this function is not against failing the suite.

Feels weird having this return a bool here, that is then funnelled into a `suite.Require().True()` in the consumer above this one.

Reply:
Doing the assertion here creates a lot of unwanted output every time the assertion fails, which I was trying to avoid.